Core/include/Acts/Seeding/SeedFilterConfig.hpp

```diff
@@ -9,12 +9,12 @@
 #pragma once
 
 #include "Acts/Definitions/Units.hpp"
+#include "Acts/Seeding/SeedfinderConfig.hpp"
```

Comment:
The problem is here. Which is also quite evident from the compilation error...

```
/usr/local/cuda/bin/nvcc -forward-unknown-to-host-compiler -DBOOST_ALL_NO_LIB -I../Plugins/Cuda/include -I../Core/include -ICore -isystem=/usr/local/include/eigen3 --extended-lambda -O3 -DNDEBUG -Xcompiler=-fPIC   -std=c++14 -MD -MT Plugins/Cuda/CMakeFiles/ActsPluginCuda2.dir/src/Seeding2/FindTriplets.cu.o -MF Plugins/Cuda/CMakeFiles/ActsPluginCuda2.dir/src/Seeding2/FindTriplets.cu.o.d -x cu -dc ../Plugins/Cuda/src/Seeding2/FindTriplets.cu -o Plugins/Cuda/CMakeFiles/ActsPluginCuda2.dir/src/Seeding2/FindTriplets.cu.o
In file included from /usr/local/include/eigen3/Eigen/Core:257:0,
                 from ../Core/include/Acts/Definitions/Algebra.hpp:15,
                 from ../Core/include/Acts/Seeding/SeedfinderConfig.hpp:11,
                 from ../Core/include/Acts/Seeding/SeedFilterConfig.hpp:12,
                 from ../Plugins/Cuda/src/Seeding2/FindTriplets.cu:19:
/usr/local/cuda/bin/../targets/x86_64-linux/include/host_defines.h:54:2: warning: #warning "host_defines.h is an internal header file and must not be used directly.  This file will be removed in a future CUDA release.  Please use cuda_runtime_api.h or cuda_runtime.h instead." [-Wcpp]
 #warning "host_defines.h is an internal header file and must not be used directly.  This file will be removed in a future CUDA release.  Please use cuda_runtime_api.h or cuda_runtime.h instead."
  ^~~~~~~
```

In Acts we use a pretty old version of both CUDA and Eigen. The two really don't like each other. :frowning:  So we had to hide all Eigen headers from the `.cu` files. But with this addition you exposed CUDA to Eigen.

You'll need to move `Acts::SeedConfirmationRange` into a separate header, from where both `SeedfinderConfig.hpp` and `SeedFilterConfig.hpp` can pick it up. With `SeedFilterConfig.hpp` not getting exposed to Eigen along the way. (It's the `Algebra.hpp` header that needs to be hidden from CUDA. `Units.hpp` you are allowed to include in that future `SeedConfirmationRange.hpp` header.)

Reply:
Done! Thank you very much @krasznaa and @konradkusiak97!